### PR TITLE
Add CI for Linux

### PR DESCRIPTION
**Pull request by omajid**
This is a followup from https://github.com/dotnet/roslyn-analyzers/pull/4419

Is there some way I can verify that CI is working before merging this?

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -78,3 +78,40 @@ jobs:
...
+    matrix:
+      Debug:
+        _configuration: Debug
+        _codeCoverage: True
+      Release:
+        _configuration: Release
+        _codeCoverage: False
+  pool:
+    name: NetCorePublic-Pool
+    queue: 'BuildPool.Ubuntu.1604.amd64.Open'
```
Comment (omajid) [inline azure-pipelines.yml:94]: I took this from dotnet/roslyn. Are there other pools available? Ubuntu 16.04 is quite old; I would prefer to use something more recent here (maybe as an alternative?).

Comment (mavasani) [reply azure-pipelines.yml:94]: @JoeRobich @dotnet/roslyn-infrastructure for guidance.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -78,3 +78,40 @@ jobs:
...
       continueOnError: true
       condition: not(succeeded())
+
+- job: Ubuntu 16.04
+  strategy:
+    maxParallel: 4
+    matrix:
+      Debug:
+        _configuration: Debug
+        _codeCoverage: True
```
Comment (mavasani) [inline azure-pipelines.yml:88]: Not sure if we need code coverage for Linux runs...

File: azure-pipelines.yml (hunk 1 above)
Comment (JoeRobich) [reply azure-pipelines.yml:92]: That is the most recent ubuntu arcade queue for now.

Comment (mavasani) [reply azure-pipelines.yml:92]: @JoeRobich What would be a good way to test this? Seems the tests are not running for this PR.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -78,3 +78,38 @@ jobs:
         ArtifactName: 'Windows Desktop Spanish'
       continueOnError: true
       condition: not(succeeded())
+
+- job: Ubuntu 16.04
```
Comment (JoeRobich) [inline azure-pipelines.yml:82]: Spaces aren't allowed in AZDO job names. This should get CI running.

```suggestion
- job: Ubuntu_16.04
```

File: azure-pipelines.yml (hunk 1 above)
Comment (JoeRobich) [reply azure-pipelines.yml:92]: A space in the job name is stopping the pipeline from running.

> Job Ubuntu 16.04 has an invalid name. Valid names may only contain alphanumeric characters and '_' and may not start with a number.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -78,3 +78,38 @@ jobs:
...
+        _configuration: Release
+  pool:
+    name: NetCorePublic-Pool
+    queue: 'BuildPool.Ubuntu.1604.amd64.Open'
+  timeoutInMinutes: 40
+
+  steps:
+    - checkout: self
+      clean: true
+    - script: ./build.sh --configuration $(_configuration) --prepareMachine
```
Comment (jmarolf) [inline azure-pipelines.yml:98]: please use

`    - script: eng\common\cibuild.sh -configuration $(_configuration) -prepareMachine`

**Comment by mavasani** [comment]
@omajid FYI that @jmarolf created https://github.com/dotnet/roslyn-analyzers/pull/4444, and it seems there are real test failures on linux that someone needs to investigate. I'll take a look a bit later today or next week if no one has got to it.

**Comment by omajid** [comment]
Oh. Thanks for that pointer. Should I close this PR, then?

I am not familiar enough with this codebase, but I can take a quick look at the unit test failures later today.

**Comment by omajid** [comment]
Closing in favour of https://github.com/dotnet/roslyn-analyzers/pull/4444

**Pull request closed without merging**